### PR TITLE
feat(api): route to fetch module by name

### PR DESCRIPTION
**Pull request by larbish**
In order to resolve nuxtlabs/nuxt.com#673, I need a route to fetch a module by name. 

This PR add a server route on `/api/modules/:name`.

I tried to use a nested router but I'm facing some issues. I will create another issue to explain my problem.

### CODE REVIEW

File: website/server/api/modules/[name].ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { sendError, createError } from 'h3'
+import { ModuleInfo } from '../../../../lib/types'
+import { fetchModuleStats } from '../modules'
+
+export default defineEventHandler(async (event) => {
+  const modules = await import('../../../../npm/modules.json').then(r => r.default || r) as ModuleInfo[]
+  const module = modules.find(m => m.name === event.context.params.name)
+
+  if (!module) {
+    return sendError(event, createError({ statusCode: 404, statusMessage: 'Module not found' }))
```
Comment (pi0) [inline website/server/api/modules/[name].ts:10]:
```suggestion
    throw createError({ statusCode: 404, statusMessage: 'Module not found' })
```

File: website/server/api/modules/[name].ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { sendError, createError } from 'h3'
```
Comment (pi0) [inline website/server/api/modules/[name].ts:1]:
```suggestion
import { createError } from 'h3'
```

**Comment by pi0** [comment]
> I am fine with it, but not sure if the API is indented to be consumed by the public. /cc @pi0 (if not, should we add an option to disable CORS in Nuxt/Nitro core?)

We can add a middleware for /api enabling CORS

File: modules/delay-hydration.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name: delay-hydration
 description: Improve your Nuxt.js v2 Google Lighthouse score by delaying hydration.
 repo: harlan-zw/nuxt-delay-hydration
-npm: nuxt-delay-hydration
+npm: mono-nuxt-delay-hydration
```
Comment (Atinux) [inline modules/delay-hydration.yml:4]:
```suggestion
npm: nuxt-nuxt-delay-hydration
```

Comment (pi0) [reply modules/delay-hydration.yml:4]: this has to be fixed in main. `repo` field needs update

**Comment by larbish** [comment]
@Atinux @pi0 @antfu I need this PR to be merged in order to resolve nuxtlabs/nuxt.com#673. Is there something I need to add ?

**Review by antfu — APPROVED** [review]

**Pull request merged**